src/Workspaces/Core/Portable/VirtualChars/VirtualCharExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.VirtualChars
+{
+    internal static class VirtualCharExtensions
+    {
+        public static string CreateString(this ImmutableArray<VirtualChar> chars)
+        {
+            var builder = PooledStringBuilder.GetInstance();
+
+            foreach (var vc in chars)
```

Comment:
Wouldn't the following remove an instance of the enumerator. 
```c#
for(var idx = 0; idx < chars.length; idx++)
{
  builder.Builder.Append(chars[idx].char);
}.

Reply:
the enumerator is a struct.  so removing it is unnecessary.

Reply:
#wontfix